includes/job_runner.h

```diff
@@ -11,7 +11,10 @@ namespace cris::core {
 
 class JobRunnerWorker;
 
-class JobRunner {
+// An object for serialized jobs. The jobs bound to the same strand object must run sequentially.
```

Comment:
Not exactly. You may add all jobs through a single strand then you serialize them all, or you may have multiple strand objects and there will be multiple sequences, or you may add some jobs without strand so that they will be concurrent with other jobs.

Reply:
```
-+-----+-+----+
 +-----+ +----+
 +-----+ +----+
-----------------
```

Reply:
What do you mean by "sausage-like"?

Reply:
As discussed offline, we may have something like this in the future:

```
// serialized by strand1
message callback:
    strand2.AddJob(spawning_job);

    // Add a condition to check if spawned jobs are complete.
    // The strand add the job on the pending queue head to the runner
    // only if the condition meets.
    //
    // We may also have other primitives, like barriers.
    strand2.AddJob(joining_job, []{ return spawned_jobs_complete; });
```

So in strand2, there will always be a spawning job first, and then a joining job waiting for completion. That is how we have sausage-like scheduling in the future.